core/commands/escrow/escrow.go

```diff
@@ -0,0 +1,151 @@
+package escrow
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/TRON-US/go-btfs/core/commands/storage"
+
+	config "github.com/TRON-US/go-btfs-config"
+	"github.com/gogo/protobuf/proto"
+	ic "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/tron-us/go-btfs-common/crypto"
+	escrowpb "github.com/tron-us/go-btfs-common/protos/escrow"
+	"github.com/tron-us/go-btfs-common/protos/ledger"
+	"google.golang.org/grpc"
+)
+
+func NewContract(configuration *config.Config, id string, payerID string, recverID string, price int64) *escrowpb.EscrowContract {
+	payerAddr := []byte(payerID)
+	recverAddr := []byte(recverID)
+	// TODO: Make escrow GuardAddress configurable
+	//authAddress := configuration.Services.GuardPubKeys[0]
+	authAddress := "guardAddress"
+	return &escrowpb.EscrowContract{
+		ContractId:       id,
+		BuyerAddress:     payerAddr,
+		SellerAddress:    recverAddr,
+		AuthAddress:      []byte(authAddress),
+		Amount:           price,
+		CollateralAmount: 0,
+		WithholdAmount:   0,
+		TokenType:        escrowpb.TokenType_BTT,
+		PayoutSchedule:   0,
+		NumPayouts:       1,
+	}
+}
+
+func NewSignedContract(contract *escrowpb.EscrowContract) *escrowpb.SignedEscrowContract {
+	return &escrowpb.SignedEscrowContract{
+		Contract: contract,
+	}
+}
+
+func NewContractRequest(configuration *config.Config, chunkInfo map[string]*storage.Chunk) (*escrowpb.EscrowContractRequest, error) {
```

Comment:
I prefer if you move all other functions to `core/escrow` instead of here. This is because `core/hub` `core/ledger` `core/guard` have all existed to perform service-specific functions outside the `core/commands`. This makes testing and also dependencies clear. The `core/escrow` functions should have 0 dependencies on `core/commands/*`.

As for this NewContractRequest, try to make what's relevant to storage stay here in `core/commands/escrow`, and call `core/escrow` as needed.

Reply:
Sure, I updated. Is it what you are expected?

Reply:
looks good